internal/pkg/agent/application/application.go

```diff
@@ -296,6 +308,86 @@ func New(
 	return coord, configMgr, varsManager, nil
 }
 
+// normalizeInstallDescriptorAtStartup will check the install descriptor checking:
+// - if we just rolled back: the update marker is checked and in case of rollback we clean up the entry about the failed upgraded install
+// - check all the entries:
+//   - verify that the home directory for that install still exists (remove what does not exist anymore)
+//   - TODO check TTLs of installs to schedule delayed cleanup while the agent is running
+//
+// This function will NOT error out, it will log any errors it encounters as warnings but any error must be treated as non-fatal
+func normalizeInstallDescriptorAtStartup(log *logger.Logger, topDir string, now time.Time, initialUpdateMarker *upgrade.UpdateMarker, rollbackSource rollbacksSource) {
+	// Check if we rolled back and update the install descriptor
+	if initialUpdateMarker != nil && initialUpdateMarker.Details != nil && initialUpdateMarker.Details.State == details.StateRollback {
+		// Reset the TTL for the current version if we are coming off a rollback
+		rollbacks, err := rollbackSource.Get()
+		if err != nil {
+			log.Warnf("Error getting available rollbacks from rollbackSource during startup check: %s", err)
+			return
+		}
+
+		// remove the current versioned home TTL marker
+		delete(rollbacks, initialUpdateMarker.PrevVersionedHome)
+		err = rollbackSource.Set(rollbacks)
+		if err != nil {
+			log.Warnf("Error removing install descriptor from installDescriptorSource during startup check: %s", err)
+			return
+		}
+	}
+
+	// check if we need to cleanup old agent installs
+	rollbacks, err := rollbackSource.Get()
+	if err != nil {
+		log.Warnf("Error getting available rollbacks during startup check: %s", err)
+		return
+	}
+
+	var versionedHomesToCleanup []string
+	for versionedHome, ttlMarker := range rollbacks {
+
+		versionedHomeAbsPath := filepath.Join(topDir, versionedHome)
+
+		if versionedHomeAbsPath == paths.HomeFrom(topDir) {
+			// skip the current install
+			log.Warnf("Found a TTL marker for the currently running agent at %s. Skipping cleanup...", versionedHome)
```

Comment:
Not sure this should be a warning, as this is almost always expected to happen, right?

Reply:
It should not be a normal case having cleaned up after a rollback in the block above (lines 320-335).
If we still have a TTL for the version of agent that is currently starting it means that something went wrong somewhere. The code will protect the current install by not trying to clean up the current install but I believe that we should at least log a warning as this will not impede the current operation but could be a symptom of something that needs fixing